### DISCUSSION TITLE
Currently planned features

### DISCUSSION THREAD

**Discussion by Cyrexxis** [Cyrexxis/RokTracker#64] [Ideas]
Hello everyone.

This discussion is for suggestions of new features and to show what is currently planned. So feel free to share your ideas.

This is the list of planned features (I will update them if I find ideas interesting):
* Option to limit the power to 150M in kingdom scan
* Preset in the GUI version of the kingdom scanner
* Custom preset that can be saved
* Upgrade alliance scanner to scan multiple stats (this will get difficult)
* Add option to run the scrips via cmd arguments

**Comment by ShawnCrocker** [comment]
Option to output CSV not Excel please. I  load a database for full kingdom history

**Reply by Cyrexxis** [reply]
Thanks for the suggestion, I think that should be possible. I will add it to the list.

**Comment by ShawnCrocker** [comment]
Thanks. I added it to my modified nikolakis1919 code, if you want my hacked and bashed sample.

**Comment by sonar0m** [comment]
your "seed scanner" can do a LOT more then Seed. examples are anything in that same format i found you can estimate 
CH 25 
mad sciance
builder
helps

so anything in that format. with some modification you could include if it were a player list. to copy the name from there. if that is required or helpful (could be optional)
this would then make a better scan for the lower seeds. so that you can create a sheet that would remove non CH 25 accounts

**Reply by Cyrexxis** [reply]
Yes, that is correct. The seed scanner works in all screens you get from the kingdom stats list. For the alliance stuff (mad scientist, etc.) there is the alliance scanner. Currently, they are very similar, but if Lilith decides to change the layout of one of those, they will not be interchangeable anymore. Also, the seed scanner can calculate the total.

As for the name copy, that only works for kingdom stuff. When you are in the alliance rankings, you cannot click on the player to open the profile and thus not copy the name. (or did I misunderstand what you mean with player list?)

Even though the seed scanner was designed to not open the  profile, it could be done optionally.

**Reply by sonar0m** [reply]
yes that's what i was talking about. you might find it use full to provide example of where or what type of things you can do with each of they types of bot's. but up to you. 
thank you sir

**Comment by ghost** [comment]
I have some simple ideas something like

-  progress bar to show the scanning progress
- Improve error handling to provide more detailed error messages and possible solutions
- notifications popup to alert users when a scan is complete or if an error occurs

**Reply by Cyrexxis** [reply]
Here are my thoughts on those points:
 - Progress bar: Yes, it could be added. Should it show the complete scan progress or the progress of the individual governors?
 - Improving error handling is also on my list. However, it is very difficult to do that when the user is manually editing a config file. For that reason, the next Version will include a UI for changing the settings. That way, at least invalid config files can be prevented.
 - Notifications: Some other person also suggested that, I have to look into that.

**Reply by ghost** [reply]
The progress bar showing the full scan progress is best imo, the individual gov progress bar is kind of unnecessary.

**Comment by Sureshot2284** [comment]
It would be nice to have a domain scanner, for example it would start on the search governor screen, the end user would provide a list (maybe CSV) of governor IDs they want to scan. Then from that screen it would get all the data for the requested governor IDs.

**Comment by leonyonz** [comment]
Hi guys, some KD have Governor location tracker bot.

Any idea how the logic is ?

**Reply by sonar0m** [reply]
are you asking how they do it or how you can do it with the existing logic. 
you can use OCR and look for the "ASSIST" and "REINFORCE" button's. once you click on the screen.
or zoom out 
![image](https://github.com/user-attachments/assets/52b1bfed-4913-4570-a260-af5321962dea)
click on the tree, then OCR for name's  or char's with in the bounds. and click just above it.

they just i think used a custom client and look for the object's in memory.

**Reply by leonyonz** [reply]
its only work when the bot account inside within same alliance.

I guess they (bot provider) do some random scroll scanning through the map, and im belive its use multiple account to mapping each city location.

Reff: https://www.youtube.com/watch?v=-bluDwZm-0g

**Comment by samuel162231** [comment]
> Hello everyone.
> 
> This discussion is for suggestions of new features and to show what is currently planned. So feel free to share your ideas.
> 
> This is the list of planned features (I will update them if I find ideas interesting):
> 
> * Option to limit the power to 150M in kingdom scan
> * Preset in the GUI version of the kingdom scanner
> * Custom preset that can be saved
> * Upgrade alliance scanner to scan multiple stats (this will get difficult)
> * Add option to run the scrips via cmd arguments

can u make the tools to convert from hall of heroes screenshots to CSV. It will helpfull
